app/views/unattended/provisioning_templates/snippet/redhat_register.erb

```diff
@@ -110,10 +110,14 @@ snippet: true
   echo "Starting the subscription-manager registration process"
 
   <% if !atomic %>
-    if [ -f /usr/bin/dnf ]; then
-      dnf -y install subscription-manager
-    else
-      yum -t -y install subscription-manager
+    # Avoid timeout accessing unreachable repo on air gapped infrastructure,
+    #  assuming subscription-manager is installed in custom packages section.
+    if [ -z "$(rpm -qa subscription-manager)" ]; then
```

Comment:
Should it instead use `rpm --query --quiet subscription-manager` and rely on the exit code?
```suggestion
    if ! rpm --query --quiet subscription-manager ; then
```